diff --git a/ant/fs/manager.py b/ant/fs/manager.py
index 62c17e7..97e8b60 100644
--- a/ant/fs/manager.py
+++ b/ant/fs/manager.py
@@ -375,6 +375,17 @@ class Application:
         response = self._get_command()
         return (response.get_serial(), response.get_data_string())
 
+    def authentication_passthrough(self):
+        self._send_command(AuthenticateCommand(
+            AuthenticateCommand.Request.PASS_THROUGH, self._serial_number))
+
+        response = self._get_command()
+        if response._get_argument("type") == AuthenticateResponse.Response.ACCEPT:
+            return response.get_data_array()
+        else:
+            raise AntFSAuthenticationException("Passthrough authentication failed",
+                                               response._get_argument("type"))
+
     def authentication_passkey(self, passkey):
         self._send_command(AuthenticateCommand(
             AuthenticateCommand.Request.PASSKEY_EXCHANGE,
